pkg/config/validation/validation.go

```diff
@@ -774,9 +774,8 @@ var ValidateSidecar = RegisterValidateFunc("ValidateSidecar",
 				if i.Tls.Mode == networking.ServerTLSSettings_ISTIO_MUTUAL || i.Tls.Mode == networking.ServerTLSSettings_AUTO_PASSTHROUGH {
 					errs = AppendValidation(errs, fmt.Errorf("configuration is invalid: cannot set mode to %s in sidecar ingress tls", i.Tls.Mode.String()))
 				}
-				protocol := protocol.Parse(i.Port.Protocol)
-				if !protocol.IsTLS() {
-					errs = AppendValidation(errs, fmt.Errorf("server cannot have TLS settings for non HTTPS/TLS ports"))
+				if i.Port.Protocol == "" {
```

Comment:
Disallow auto to avoid split into http and tcp listeners, in which case there will be ambigous filter chain matches

Reply:
This is something confusing. before, we know a port is tls should have tls setting, but now a non tls port can have tls setting.

Reply:
Does it make more sense to come up with something instead for http2 + tls? E.g. `H2` or `HTTPS2`?

Reply:
I think it makes much sense, long long ago there is such issue https://github.com/istio/istio/issues/16282